Makefile

```diff
@@ -70,87 +109,6 @@ uninstall::
-export LINUX_DISTRO_RELEASE := $(shell lsb_release -r -s || true)
-prepare:: linux-prepare
-endif # $(OS_NAME)=Linux
-
-prepare:: prepare-paths prepare-msbuild
-
-linux-prepare::
-	BINFMT_MISC_TROUBLE="cli win" \
-	BINFMT_WARN=no ; \
-	for m in ${BINFMT_MISC_TROUBLE}; do \
-		if [ -f "/proc/sys/fs/binfmt_misc/$$m" ]; then \
-			BINFMT_WARN=yes ; \
-		fi ; \
-	done ; \
-	if [ "x${BINFMT_WARN}" = "xyes" ]; then \
-		cat Documentation/binfmt_misc-warning-Linux.txt ; \
-	fi; \
-	if [ -f build-tools/scripts/dependencies/linux-prepare-$(LINUX_DISTRO)-$(LINUX_DISTRO_RELEASE).sh ]; then \
-		sh build-tools/scripts/dependencies/linux-prepare-$(LINUX_DISTRO)-$(LINUX_DISTRO_RELEASE).sh $(LINUX_DISTRO_RELEASE); \
-	elif [ -f build-tools/scripts/dependencies/linux-prepare-$(LINUX_DISTRO).sh ]; then \
-		sh build-tools/scripts/dependencies/linux-prepare-$(LINUX_DISTRO).sh $(LINUX_DISTRO_RELEASE); \
-	fi
-
-# $(call GetPath,path)
-GetPath   = $(shell $(MSBUILD) $(MSBUILD_FLAGS) /p:DoNotLoadOSProperties=True /nologo /v:minimal /t:Get$(1)FullPath build-tools/scripts/Paths.targets | tr -d '[[:space:]]' )
-
-MSBUILD_PREPARE_PROJS = \
-	src/mono-runtimes/mono-runtimes.csproj \
-	src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.csproj
-
-prepare-deps: prepare-cmake-mingw-toolchain
-	git submodule update --init --recursive
-	./build-tools/scripts/generate-os-info Configuration.OperatingSystem.props
-	mkdir -p bin/Build$(CONFIGURATION)
-	$(call MSBUILD_BINLOG,prepare-deps) build-tools/dependencies/dependencies.csproj
-
-#
-# $(1): output file name
-#
-define create_cmake_toolchain
-prepare-cmake-mingw-toolchain:: bin/Build$(CONFIGURATION)/$(1)
-
-bin/Build$(CONFIGURATION)/$(1): build-tools/scripts/$(1).in
-	mkdir -p $$(dir $$@)
-	sed -e 's;@HOMEBREW_PREFIX@;$$(HOMEBREW_PREFIX);g' < $$< > $$@
-endef
-
-$(eval $(call create_cmake_toolchain,mingw-32.cmake))
-$(eval $(call create_cmake_toolchain,mingw-64.cmake))
-
-prepare-external: prepare-deps
-	nuget restore $(NUGET_VERBOSITY) $(SOLUTION)
-	nuget restore $(NUGET_VERBOSITY) Xamarin.Android-Tests.sln
-	(cd external/xamarin-android-tools && make prepare CONFIGURATION=$(CONFIGURATION))
-	(cd $(call GetPath,JavaInterop) && make prepare CONFIGURATION=$(CONFIGURATION) JI_MAX_JDK=8)
-	(cd $(call GetPath,JavaInterop) && make bin/Build$(CONFIGURATION)/JdkInfo.props CONFIGURATION=$(CONFIGURATION) JI_MAX_JDK=8)
-
-prepare-bundle: prepare-external
-	$(call MSBUILD_BINLOG,prepare-bundle) build-tools/download-bundle/download-bundle.csproj
-	$(call MSBUILD_BINLOG,prepare-restore) $(MSBUILD_FLAGS) tests/Xamarin.Forms-Performance-Integration/Xamarin.Forms.Performance.Integration.csproj /t:Restore
-
-prepare-props: prepare-bundle
-	cp $(call GetPath,JavaInterop)/external/Mono.Cecil* "$(call GetPath,MonoSource)/external"
-	cp "$(call GetPath,JavaInterop)/product.snk" "$(call GetPath,MonoSource)"
-	cp build-tools/scripts/Configuration.Java.Interop.Override.props external/Java.Interop/Configuration.Override.props
```

Comment:
I think the current test failures might be related to this line. We should still override the java-interop's props to make it use the mono's cecil.

Reply:
We still do it, [see here](https://github.com/grendello/xamarin-android/blob/xa/bootstrap/build-tools/xaprepare/xaprepare/Steps/Step_PrepareProps.cs#L34-L39)